### PR TITLE
Prepare repo for .NET 8 

### PR DESCRIPTION
**Pull request by andreasohlund**
Downstream follow-up to [preparing for .NET 8](https://github.com/Particular/NServiceBus/pull/6810)

### CODE REVIEW

**Comment by andreasohlund** [comment]
Tested manually with the emulator and all seems to work as expected

File: src/NServiceBus.AzureFunctions.Worker.ServiceBus/Serverless/TransportWrapper/ServerlessTransport.cs [hunk 1]
```diff
@@ -42,9 +43,7 @@ class ServerlessTransport : TransportDefinition
         }
 
 #pragma warning disable CS0672 // Member overrides obsolete member
-#pragma warning disable CS0618 // Type or member is obsolete
-        public override string ToTransportAddress(QueueAddress address) => baseTransport.ToTransportAddress(address);
-#pragma warning restore CS0618 // Type or member is obsolete
+        public override string ToTransportAddress(QueueAddress address) => throw new NotImplementedException();
```
Comment (kentdr) [inline src/NServiceBus.AzureFunctions.Worker.ServiceBus/Serverless/TransportWrapper/ServerlessTransport.cs:46]: We changed `ToTransportAddress` to [virtual in core](https://github.com/Particular/NServiceBus/pull/6819/files#diff-606b38176cb5c2adf4e0855f68c3d3e796031b215246ffec036f61af93c81b64R213) so it does not need to be overridden in the downstream packages.  I realize you'd have to change AzureServiceBus first, but we may want to wait to merge this until AzureServiceBus is bumped to alpha.2 of core.

Comment (andreasohlund) [reply src/NServiceBus.AzureFunctions.Worker.ServiceBus/Serverless/TransportWrapper/ServerlessTransport.cs:46]: Agreed, made this a draft pr until I have that fixed

**Comment by andreasohlund** [comment]
Should be good to go now

File: src/NServiceBus.AzureFunctions.Worker.SourceGenerator/TriggerFunctionGenerator.cs [hunk 2]
```diff
@@ -10,23 +9,23 @@
     public class TriggerFunctionGenerator : ISourceGenerator
     {
         internal static readonly DiagnosticDescriptor InvalidEndpointNameError = new DiagnosticDescriptor(
-            id: "NSBWFUNC 001",
+            id: "NSBWFUNC001",
```
Comment (andreasohlund) [inline src/NServiceBus.AzureFunctions.Worker.SourceGenerator/TriggerFunctionGenerator.cs:12]: Seems like MS is now validating the IDs and they can't have whitespace in them

Comment (bording) [reply src/NServiceBus.AzureFunctions.Worker.SourceGenerator/TriggerFunctionGenerator.cs:12]: 👍 Part of updating Microsoft.CodeAnalysis.CSharp. For some reason I didn't see this pop up locally even though I built after updating the package.

**Review by kentdr — APPROVED** [review]

**Comment by andreasohlund** [comment]
Rebased and ready for final review

**Review by bording — APPROVED** [review]

**Pull request merged**